File: Frontend/src/utils/billSettings.js
export const defaultSettings = {
  headerText: "TechSentinals Optical Store",
  subHeaderText: "Tax Invoice / Receipt",
  contactEmail: "info@techsentinals.com",
  contactPhone: "9876543210",
  address: "123, Optical Plaza, Main Road",
  city: "Mumbai",
  state: "Maharashtra",
  pincode: "400001",
  gstNumber: "27AAAAA1111A1Z1",
  footerText: "Thank you for shopping with us! Please visit again.",
  themeColor: "#10B981", // default emerald
  showPrescription: true,
  showGst: true,
  logo: null, // base64 string
  qrCode: null, // base64 string
};

export const getBillTemplateSettings = (storeId) => {
  const saved = localStorage.getItem(`bill_template_settings_${storeId || 'default'}`);
  if (saved) {
    try {
      return { ...defaultSettings, ...JSON.parse(saved) };
    } catch {
      return defaultSettings;
    }
  }
  return defaultSettings;
};

export const saveBillTemplateSettings = (storeId, settings) => {
  localStorage.setItem(`bill_template_settings_${storeId || 'default'}`, JSON.stringify(settings));
};

File: Backend/schemas/bill_settings.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime

class BillSettingsBase(BaseModel):
    header_text: Optional[str] = "Optical Store"
    sub_header_text: Optional[str] = "Tax Invoice / Receipt"
    address: Optional[str] = None
    contact_email: Optional[str] = None
    contact_phone: Optional[str] = None
    gst_number: Optional[str] = None
    logo: Optional[str] = None
    qr_code: Optional[str] = None
    show_prescription: bool = True
    show_gst: bool = True
    theme_color: Optional[str] = "#0A0F1F"
    footer_text: Optional[str] = "Thank you for your business!"

class BillSettingsUpdate(BillSettingsBase):
    pass

class BillSettingsResponse(BillSettingsBase):
    id: int
    store_id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
